support/jakarta-ee/src/main/java/org/apache/shiro/ee/cdi/KeyGen.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.shiro.ee.cdi;
+
+import java.nio.charset.StandardCharsets;
+import java.util.function.Supplier;
+import org.apache.shiro.crypto.cipher.AesCipherService;
+import org.apache.shiro.mgt.AbstractRememberMeManager;
+import org.apache.shiro.web.mgt.DefaultWebSecurityManager;
+import org.omnifaces.util.Beans;
+import org.omnifaces.util.Lazy;
+import static org.omnifaces.util.Utils.isBlank;
+
+/**
+ * Shiro cipher key generator
+ *
+ * @author lprimak
+ */
+public class KeyGen {
+    private final Lazy<AesCipherService> cipherService = new Lazy<>(AesCipherService::new);
```

Comment:
This is very elegant and easy to read, but a static inner `Holder` class would also do.

Reply:
Trying to be concise and limit the lines of code here

Reply:
Guess it is a question, if one wants to replace OF or not ;-)